File: src/contests/cg/puzzle/teds-compiler.ts
// 🎮 CodinGame Puzzle - teds-compiler
// https://www.codingame.com/training/easy/teds-compiler

const line = readline()
let depth = 0
let best = 0
for (let i = 0; i < line.length; i++) {
  if (line[i] === "<") depth++
  else depth--
  if (depth < 0) break
  if (depth === 0) best = i + 1
}
console.log(best)

File: src/contests/cg/puzzle/ocaml/teds-compiler.ml
(* 🎮 CodinGame Puzzle - teds-compiler *)
(* https://www.codingame.com/training/easy/teds-compiler *)

let () =
  let line = String.trim (read_line ()) in
  let balance = ref 0 and best = ref 0 in
  (try
     for i = 0 to String.length line - 1 do
       if line.[i] = '<' then incr balance else decr balance;
       if !balance < 0 then raise Exit;
       if !balance = 0 then best := i + 1
     done
   with Exit -> ());
  Printf.printf "%d\n" !best
